lib/assert.js

```diff
@@ -103,8 +104,21 @@ x.throws = function (fn, err, msg) {
 			}
 		}, err, msg);
 
+		if (!isError(result)) {
+			throw result;
+		}
+
 		return result;
 	} catch (err) {
+		if (!isError(err)) {
+			var actual = Object.prototype.toString.call(err);
+			var expected = '[object Error]';
+			var message = 'Error `' + err + '` is not of type `' + expected + '` but of type `' + actual + '`.';
```

Comment:
The expectation depends on the original `err` argument, be it a constructor, string, regular expression or validation function.

I'm starting to think we need to reimplement `assert.throws()` here. It's too hard to format our assertion error without duplicating its internals.